File: rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/fresco/CustomImageFormatConfigurator.java
package chat.rocket.android.widget.fresco;

import android.support.annotation.Nullable;
import com.facebook.drawee.backends.pipeline.DraweeConfig;
import com.facebook.imagepipeline.decoder.ImageDecoderConfig;

/**
 * Helper class to add custom decoders and drawable factories.
 * See: https://github.com/facebook/fresco/blob/master/samples/showcase/src/main/java/com/facebook/fresco/samples/showcase/CustomImageFormatConfigurator.java
 */
public class CustomImageFormatConfigurator {

  @Nullable
  public static ImageDecoderConfig createImageDecoderConfig() {
    ImageDecoderConfig.Builder config = ImageDecoderConfig.newBuilder();
    config.addDecodingCapability(SvgDecoder.SVG_FORMAT, new SvgDecoder.SvgFormatChecker(), new SvgDecoder.Decoder());
    return config.build();
  }

  public static void addCustomDrawableFactories(DraweeConfig.Builder draweeConfigBuilder) {
    draweeConfigBuilder.addCustomDrawableFactory(ColorImage.createDrawableFactory());
    draweeConfigBuilder.addCustomDrawableFactory(new SvgDecoder.SvgDrawableFactory());
  }
}

File: app/src/main/java/chat/rocket/android/app/utils/CustomImageFormatConfigurator.kt
package chat.rocket.android.app.utils

import com.facebook.drawee.backends.pipeline.DraweeConfig
import com.facebook.imagepipeline.decoder.ImageDecoderConfig

/**
 * Utility class to add custom decoders and drawable factories.
 *
 * @see {https://github.com/facebook/fresco/blob/master/samples/showcase/src/main/java/com/facebook/fresco/samples/showcase/CustomImageFormatConfigurator.java}
 */
object CustomImageFormatConfigurator {

    fun createImageDecoderConfig() : ImageDecoderConfig {
        return ImageDecoderConfig.newBuilder()
                .addDecodingCapability(SvgDecoder.svgFormat, SvgDecoder.SvgFormatChecker(), SvgDecoder.Decoder())
                .build()
    }

    fun addCustomDrawableFactories(draweeConfigBuilder: DraweeConfig.Builder) {
        // We always add the color drawable factory so that it can be used for image decoder overrides.
        draweeConfigBuilder.addCustomDrawableFactory(ColorImage.createDrawableFactory())
        draweeConfigBuilder.addCustomDrawableFactory(SvgDecoder.SvgDrawableFactory())
    }
}